### ISSUE TITLE
Feature request - Custom defined relationship memory

### ISSUE THREAD

**Issue by erikfrye** [pods-framework/pods#3546]
In our use of pods we have a couple shared relationship fields across multiple custom post type pods. The values in these simple relationships are the same and would be helpful if there was a memory of sorts, where updating the relationship values in one pod would update the others.

**Comment by jimtrue** [comment]
You can set your relationships between two Pods as Bidirectional on a specific field. That ensures that changes you make on one side are reflected on the other side of the relationship.

Is that what you mean?

**Comment by erikfrye** [comment]
Hi Jim,

Maybe this example will help.

I have a property in pods that helps us set a specific CSS position property we use when displaying an item in the post.

It's relationship values are

top-left
top-right
bottom-left
bottom-right

This relationship is used in many different post types created in pods.

Say next month I want to add a new option to this list, eg. center-center

I will need to go into each pod that uses this simple relationship and update the list of items. 

What I was trying to get at was a way that these values in the relationships could be reused so updating one would update all pods using those same values.

Hope that clears things up.

Thanks,
Erik

**Comment by jimtrue** [comment]
Ah, you wanted to be able to save the values for the Simple Custom Defined List to a different location for re-use, in much the same way that we have the US States, Countries, etc. as predefined lists, correct?

**Comment by erikfrye** [comment]
Yes, a custom predefined list. You said it much better!

**Comment by jimtrue** [comment]
Yep, that actually makes more sense. You could use a relationship to a  post type as a workaround to that at present, but that's a lot of overkill just for a post_title ;)

I've flagged this as a feature/enhancement. @sc0ttkclark wondering if there might be a filter they can hook into to create their own custom defined lists?

**Comment by jimtrue** [comment]
We don't have an Admin UI around this, but if you're not allergic to code, you can add these yourself using our pods_register_related_object function:
[](http://pods.io/docs/code/general-functions/pods-register-related-object/)

We have some sample code I'll be adding to that page, but for right now:
```php
$options = array(
    'group' => 'Special Relationships', // Whatever you want the selection group to be, defaults to label
    'simple' => false, // Whether this field is related by strings or integer IDs, integer IDs get stored in wp_podsrel, strings are stored in the field itself either serialized (meta-based) or json encoded (table-based)
    'data' => array( // Custom define the items to select from manually
        1 => 'Gravity Form 1',
        2 => 'Gravity Form 2'
    ),
    'data_callback' => 'get_custom_gravity_forms_list', // Provide a callback function to call to define the data (instead of setting 'data' above)
    'value_to_label_callback' => 'get_custom_gravity_forms_list', // Provide a callback function to define the data when called through PodsField_Pick::value_to_label
    'simple_value_callback' => 'get_custom_gravity_forms_list' // Provide a callback function to define the data when called through PodsField_Pick::simple_value
);

pods_register_related_object( 'gravity-form', 'Gravity Forms', $options );
```

**Comment by erikfrye** [comment]
Thanks I will give this a try.

On Wed, May 25, 2016 at 4:15 PM, Jim True <notifications@github.com> wrote:

> We don't have an Admin UI around this, but if you're not allergic to code,
> you can add these yourself using our pods_register_related_object function:
> <http://pods.io/docs/code/general-functions/pods-register-related-object/>
>
> We have some sample code I'll be adding to that page, but for right now:
>
> $options = array(    'group' => 'Special Relationships', // Whatever you want the selection group to be, defaults to label    'simple' => false, // Whether this field is related by strings or integer IDs, integer IDs get stored in wp_podsrel, strings are stored in the field itself either serialized (meta-based) or json encoded (table-based)    'data' => array( // Custom define the items to select from manually        1 => 'Gravity Form 1',        2 => 'Gravity Form 2'    ),    'data_callback' => 'get_custom_gravity_forms_list', // Provide a callback function to call to define the data (instead of setting 'data' above)    'value_to_label_callback' => 'get_custom_gravity_forms_list', // Provide a callback function to define the data when called through PodsField_Pick::value_to_label    'simple_value_callback' => 'get_custom_gravity_forms_list' // Provide a callback function to define the data when called through PodsField_Pick::simple_value);pods_register_related_object( 'gravity-form', 'Gravity Forms', $options );
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pods-framework/pods/issues/3546#issuecomment-221693620>
>



-- 
*Erik Frye*
erikfrye.com
@erikfrye <http://twitter.com/erikfrye>
